mint-client/src/bin/clientd/main.rs

```diff
@@ -33,43 +25,153 @@ struct Opts {
     workdir: PathBuf,
 }
 
+/*
+#[derive(Debug, Deserialize, Serialize)]
+struct InfoResponse {
+    total : serde_json::Value,
+    coins : serde_json::Value,
+}
+*/
+#[derive(Serialize)]
+struct InfoResponse {
+    total : CoinTotal,
+    coins : Vec<CoinGrouped>,
+}
+#[derive(Serialize)]
+struct CoinTotal{
+    coin_count : usize,
+    amount : Amount,
+}
+#[derive(Serialize)]
+struct CoinGrouped{
+    amount : usize,
+    tier : u64
+}
+#[derive(Serialize)]
+struct SpendResponse {
+    token : String,
+}
+
+#[derive(Serialize)]
+struct ReissueResponse {
+    id : String,
+    fetched : Vec<TransactionId>,
+}
+
+#[derive(Deserialize,Serialize, Debug)]
+struct ReqBody<T> {
+    //I dont like this
+    value : T
+}
+
+impl InfoResponse {
+    fn new(coins: Coins<SpendableCoin>) -> Self {
+        let info_total =  CoinTotal { coin_count: coins.coin_count(), amount: coins.amount() };
+        let info_coins : Vec<CoinGrouped> = coins.coins.iter()
+            .map(|(tier, c)| CoinGrouped { amount : c.len(), tier : tier.milli_sat})
+            .collect();
+        InfoResponse { total : info_total, coins : info_coins }
+    }
+}
+
+impl ReissueResponse {
+    async fn new(mint_client :&MintClient, coins:Coins<SpendableCoin>) -> Self {
+        let mut rng = rand::rngs::OsRng::new().unwrap();
+        let id = mint_client.reissue(coins, &mut rng).await.unwrap();
+
+        let  id = id.to_hex();
+        let  fetched : Vec<TransactionId>= mint_client.fetch_all_coins().await.unwrap();
+        let res = ReissueResponse {
+            id,
+            fetched,
+        };
+        res
+    }
```

Comment:
Not sure if the reissuance logic should reside in the constructor of the response.

Reply:
Agreed. The multiple `unwrap`s happening in an infallible function are not a great approach.

Reply:
Understood. Carry on.